### PR TITLE
feat: `w3 key create`

### PR DESCRIPTION
**Pull request by olizilla**
Print a new key pair. Does not change your current signing key.

Bring the feature from [`ucan-key`](https://github.com/olizilla/ucan-key) into w3cli

```shell
$ w3 key create
# did:key:z6Mkne1JnNWCvUgxsdo552FW8jRN78uxbpCnBmDiPmf1koYc
MgCZ5ftHm9e0FnNtFXA2sMZgcf97GO4vTaybGEfkBp/KR5+0BeZ44TKEOwUjbLqvFFs2hpFxD++IAEEL+3ySROyDsSRk=

$ w3 key create --json | jq
{
  "did": "did:key:z6Mkt7urpgJKK38cEyB1uMGBzfdn6sZN2daB5pr75DZbgahZ",
  "key": "MgCYj0pKJn5uf+4yr2DApB8idYtDY+9yhEYH64Po0uaj+R+0Byw7B8cSYrO4//5PqgWodYq8PMI/fSWZ1xdvdw15bmAw="
}
```

see: https://github.com/web3-storage/w3cli/issues/154#issuecomment-1885042715

License: MIT

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,47 +25,49 @@ w3 space register # defaults to registering you with web3.storage
...
+  - [`w3 space use`](#w3-space-use-did)
+  - [`w3 space info`](#w3-space-info)
+- Capability management
+  - [`w3 delegation create`](#w3-delegation-create-audience-did)
+  - [`w3 delegation ls`](#w3-delegation-ls)
+  - [`w3 delegation revoke`](#w3-delegation-revoke-delegation-cid)
+  - [`w3 proof add`](#w3-proof-add-proofucan)
+  - [`w3 proof ls`](#w3-proof-ls)
+- Key management
+- [`w3 key create`](#w3-key-create)
```
Comment (alanshaw) [inline README.md:61]:
```suggestion
  - [`w3 key create`](#w3-key-create)
```

File: README.md [hunk 2]
```diff
@@ -126,14 +128,14 @@ List spaces known to the agent.
...
 capabilities to the currently logged in account. If you are authorized against
 more than one account you'll need to pass the `--email` option to specify which account to
 register the space with.
 
-* `--email` The email address of the account to associate this space with.
-* `--provider` The storage provider to associate with this space.
-> By registering your w3up beta Space with [web3.storage](http://web3.storage/), you agree to the w3up beta [Terms of Service](https://console.web3.storage/terms). Until the beta period is over and this migration occurs, uploads to w3up will not appear in your web3.storage account (and vice versa), even if you register with the same email.
+- `--email` The email address of the account to associate this space with.
+- `--provider` The storage provider to associate with this space.
+  > By registering your w3up beta Space with [web3.storage](http://web3.storage/), you agree to the w3up beta [Terms of Service](https://console.web3.storage/terms). Until the beta period is over and this migration occurs, uploads to w3up will not appear in your web3.storage account (and vice versa), even if you register with the same email.
```
Comment (alanshaw) [inline README.md:138]:
```suggestion
```

File: bin.js [hunk 3]
```diff
@@ -294,6 +295,12 @@ cli
   .describe('Get filecoin information for given PieceCid.')
   .action(filecoinInfo)
 
+cli
+  .command('key create')
+  .describe('Print a new key pair. Does not change your current signing key')
```
Comment (alanshaw) [inline bin.js:300]:
```suggestion
  .describe('Generate and print a new ed25519 key pair. Does not change your current signing key.')
```

File: index.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ import {
   startOfLastMonth,
 } from './lib.js'
 import * as ucanto from '@ucanto/core'
+import { ed25519, RSA } from '@ucanto/principal'
```
Comment (alanshaw) [inline index.js:22]: `RSA` unused?

File: index.js [hunk 5]
```diff
@@ -572,3 +588,17 @@ async function* getSpaceUsageReports(client, period) {
...
+ * @param {{ json: boolean }} options
+ */
+export async function createKey({ json }) {
+  const signer = await ed25519.generate()
+  const key = ed25519.format(signer)
+  if (json) {
+    console.log(dagJSON.stringify(signer.toArchive()))
+  } else {
+    console.log(`# ${signer.did()}`)
+    console.log(`${key}`)
```
Comment (alanshaw) [inline index.js:602]: Is it already a string?
```suggestion
    console.log(key)
```

**Review by alanshaw — CHANGES_REQUESTED** [review]

**Comment by olizilla** [comment]
I bow deeply in apology for the prettier noise. tis an obnoxious and persistent bot, and I let my guard down. As the harm is now done, I leave it in this PR so others do not suffer the same semantically-meaningless, unrelated change fate.

**Review by alanshaw — APPROVED** [review]

**Comment by alanshaw** [comment]
I merge this so I can do some other readme cleanup.

**Pull request merged**